src/fund_my_watcard/main.py

```diff
@@ -128,5 +135,58 @@ def main():
         if query_yes_no(WILL_RESET_YOUR_CONFIG_FILE_WARNING, "no"):
             reset_config_file()
 
+    elif args.valid:
+        if os.path.isfile(CONFIG_FILE_PATH):
+            with open(CONFIG_FILE_PATH) as f:
+                _config = json.load(f)
+        else:
+            report_error(CAN_NOT_FIND_CONFIG_FILE)
+
+        if _config["encrypted"] == "True":
+            report_warning(CONFIG_FILE_HAS_BEEN_ENCRYPTED)
+        else:
+            if re.fullmatch('[!@#$%^&*(),.?":{}|<>]', _config["userName"]):
+                report_error("userName has special characters, please remove them.")
+
+            if re.fullmatch("([a-zA-Z]+ +)", _config["ordName"]):
+                report_error("ordName has characters other than letters, please remove them.")
+
+            if len(_config["phoneNumber"].replace(" ", "")) != 10:
+                report_error("phoneNumber has too few or too many characters, please check it.")
+
+            if not re.fullmatch(
+                "[ABCEGHJKLMNPRSTVXY][0-9][ABCEGHJKLMNPRSTVWXYZ] ?[0-9][ABCEGHJKLMNPRSTVWXYZ][0-9]",
+                _config["ordPostalCode"],
+            ):
+                report_error("ordPostalCode is not a valid postal code, please check it.")
+
+            if not re.fullmatch("([a-zA-Z]+)", _config["ordCity"]):
+                report_error("ordCity has characters other than letters, please remove them.")
+
+            if not re.fullmatch("([^@]+@[^@]+\\.[^@]+)", _config["ordEmailAddress"]):
+                report_error("ordEmailAddress is not a valid email address, please check it.")
+
+            if _config["paymentMethod"] != "CC":
+                report_error("paymentMethod is not CC, please check it.")
+
+            if re.fullmatch("([a-zA-Z]+ +)", _config["trnCardOwner"]):
+                report_error("trnCardOwner has characters other than letters, please remove them.")
+
+            if not re.fullmatch("(MC|VI|PV|MD|AM)", _config["trnCardType"]) and len(_config["trnCardType"]) != 2:
```

Comment:
Why we need to check the length of `trnCardType`? Is checking `fullmatch` not enough?

Reply:
I think it's my regex that's bad. Because the current regex captures all characters M,C,V,I,P,D,A but not the strings "MC", "VI", "PV" etc... I will need to fix this regex to remove the check for `trnCardType` length.